driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java

```diff
@@ -169,50 +157,45 @@ public void selectServerAsync(final ServerSelector serverSelector, final Operati
     public ClusterDescription getDescription() {
         isTrue("open", !isClosed());
 
-        try {
-            CountDownLatch currentPhase = phase.get();
-            ClusterDescription curDescription = description;
+        boolean selectionFailureLogged = false;
+        ClusterDescription currentDescription;
 
-            boolean selectionFailureLogged = false;
+        StartTime startTime = StartTime.now();
+        Timeout timeout = startServerSelectionTimeout(startTime);
 
-            long startTimeNanos = System.nanoTime();
-            long curTimeNanos = startTimeNanos;
-            long maxWaitTimeNanos = getMaxWaitTimeNanos();
+        while (true) {
+            CountDownLatch currentPhaseLatch = phase.get();
+            currentDescription = description;
 
-            while (curDescription.getType() == ClusterType.UNKNOWN) {
+            if (currentDescription.getType() != ClusterType.UNKNOWN) {
```

Comment:
Does this check then allows for a `0` value in server selection. If ones available return it else timeout?

Reply:
I made a small change to bring this getDescription in line with selectServer, but basically yes there are only two paths out of the loop. Either we throw a timeout, or we return something, and here the check is for the cluster type becoming known.